apis/medusa/v1alpha1/medusa_types.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+type Storage struct {
+	// The storage backend to use for the backups.
+	// Should be either of "local", "google_storage", "azure_blobs" or "s3".
+	StorageProvider string `json:"storageProvider,omitempty"`
+
+	// Kubernetes Secret that stores the key file for the storage provider's API.
+	// If using 'local' storage, this value is ignored.
+	// +optional
+	StorageSecretRef string `json:"storageSecretRef,omitempty"`
+
+	// The name of the bucket to use for the backups.
+	BucketName string `json:"bucketName,omitempty"`
+
+	// Name of the top level folder in the backup bucket.
+	// If empty, the cluster name will be used.
+	// +optional
+	Prefix string `json:"prefix,omitempty"`
+
+	// Maximum backup age that the purge process should observe.
+	// +optional
+	MaxBackupAge int `json:"maxBackupAge,omitempty"`
+
+	// Maximum number of backups to keep (used by the purge process).
+	// Default is unlimited.
```

Comment:
How do you represent "unlimited"? Is it zero, -1, ...?

More generally, think twice about whether to use pointers or not for a field. If not using pointers, the field will take the type's zero value by default. For numbers, this means that fields will be 0 by default. However if 0 is a valid value, then there is no way to distinguish "value is absent" from "value is zero". In such cases it's necessary to use pointers.

Reply:
It's zero, I've set a default using the kubebuilder annotation.